Update formatting for branch names

mattzollinhofer opened the pull request:
Resolves #688 which reports an issue when there is a question mark in the
method name. git spec states that question marks are not allowed. I took
more conservative approach and only allowed numbers, letters, and
underscores.

I also converted periods to pounds:
  `Payment.charge` -> `Payment#charge`

schneems commented:
Thanks a ton for this!

In the shell anything after the pound will be interpreted as a comment. I think we need another character or keeping the period seems fine.

Could you move the logic into a model or helper and then add a test?

mattzollinhofer commented:
Ha, good point about pound in shell, wasn’t thinking. I was only changing it because officially branch names also can’t have two periods in a row (‘git diff invalid..branchname..master’) and got cute with the pound:-). I’d lean towards just replacing periods like everything else and only allowing letters and numbers but I’d be happy to leave them and hope there no two in a row.  Preference?

Yeah I’ll pull that out and get some tests around it too.

mattzollinhofer commented:
I updated and removed periods, let me know if you'd prefer leaving them in.

I also decided to make a little poro rather than moving into a model or helper. I've been moving towards small objects like this in my dev work, but I know not everyone prefers that. So just say the word and I'll pull the method into a model or helper.

schneems commented:
Awesome! Thanks!
